### PR TITLE
Cart wishlist functionalities

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
I have Implemented the cart fetching and wish-list fetching.
Please review my -
components/toast component
reducerandcontext/authState
pages/authenticationPages/dataFetchingAndAuthentication

Thank you for reviewing

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -7,18 +7,34 @@ import Mockman from "mockman-js";
...
 function App() {
+const {authState,authDispatch }=useAuthorization();
+useEffect(()=>{
+ 
+const timeoutID= setTimeout(()=> {authDispatch({
+        type: "TOAST",
+        payload: { display:"none", message: "", type: "" },
+   
+      });clearTimeout(timeoutID)},2000)
+},[authState.toast.display])
```
Comment (anik31) [inline src/App.js:22]: appreciation - nice implementation of toast

File: src/components/DifferentCards/cartsCard/HorizontalCard.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+//       <span style={{ width: "5rem", height: "5rem" }}>
+//         <img src={image} alt={alt} />
+//       </span>
+//       <p className="toast-text">{name}</p>
+//       <p className="text relative">
+//         Price :<span className="margin-l-1">{price}</span>
+//       </p>
+//     </div>
+//   );
+// };
```
Comment (anik31) [inline src/components/DifferentCards/cartsCard/HorizontalCard.jsx]: suggestion - avoid pushing commented code

File: src/components/DifferentCards/cartsCard/HorizontalCard.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            <button
+              className="btn "
+              onClick={() => incrementProduct(wholeItem._id)}
+            >
+              +
+            </button>
+            <button
+              disabled={quantity === 1 ? true : false}
+              className="btn "
+              onClick={() => decreaseProduct(wholeItem._id)}
```
Comment (anik31) [inline src/components/DifferentCards/cartsCard/HorizontalCard.jsx:50]: appreciation - I like how you have abstracted out the functions

File: src/components/authentication/LogInForm.jsx [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+  const { logInHandler } = useUserDetails();
   function clickHandler(e) {
     //  to prevent initial refreshing of the page
     e.preventDefault();
-    authDispatch({ type: "LOG_IN", payload: details });
+
+    logInHandler(details.email, details.password);
+
     console.log("triggered");
   }
```
Comment (anik31) [inline src/components/authentication/LogInForm.jsx:22]: suggestion - remove console.log

File: src/components/authentication/LogInForm.jsx [hunk 5]
```diff
@@ -28,13 +32,25 @@ function LogInForm() {
         onChange={(e) => setDetails({ ...details, email: e.target.value })}
       />
       <input
-        type="password"
+        type={!viewPassword ? "password" : "text"}
         name="password"
         className="form-input"
         placeholder="Enter Your Password"
         onChange={(e) => setDetails({ ...details, password: e.target.value })}
```
Comment (anik31) [inline src/components/authentication/LogInForm.jsx:39]: suggestion - you can add value attribute & assign it with state variable to make it a controlled component

File: src/components/toast/Toast.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+function Toast({ type, message, display = "none" }) {
+  const style =
+    type === "normal"
+      ? "alert"
+      : type === "error"
+      ? "alert error"
+      : type === "success"
+      ? "alert success"
+      : null;
+
```
Comment (anik31) [inline src/components/toast/Toast.jsx:14]: appreciation - good that you have added all types of alerts

File: src/pages/contextsAndReducer/CartManagementProvider.jsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      case "REMOVE_FROM_CART": {
+        return { ...state, cart: action.payload };
+      }
+
+      case "INCREASE_QUANTITY": {
+        return { ...state, cart: action.payload };
+      }
+
+      case "DECREASE_QUANTITY": {
+        return { ...state, cart: action.payload };
```
Comment (anik31) [inline src/pages/contextsAndReducer/CartManagementProvider.jsx]: suggestion - the return value of all these cases are the same, so you can replace them with a single case "SET_CART"

**Review by anik31 — COMMENTED** [review]
Good work @Dev-AkshatGupta , everything looks good to me.

suggestion - you can divide larger files into smaller ones. Also youcan add images or video in PR description

**Review by AshutoshM29 — COMMENTED** [review]
Nice Work!!, you can merge the code

Things I like about your code:
- Good use of abbreviation for all files.
- Breaking of code in atomic functions.
- Code is readable and understandable.
- File structure

**Pull request merged**